src/hooks/useUserInfo.tsx

```diff
@@ -0,0 +1,15 @@
+import { useSelector } from "react-redux";
+import { checkTokenValidity } from "../components/features/tokenValidityCheck";
+import { RootState } from "../store";
+
+const useUserInfo = () => {
+  let isLoggedIn = true;
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const token = useSelector((state: RootState) => state.user.userToken);
+
+  return !checkTokenValidity(token)
```

Comment:
If the token is not valid, you should also not return the token here, instead undefined or null. Also use` if` here, not inline ternary as it is hard to read.